scripts/calculateMinichefOwed.ts

```diff
@@ -0,0 +1,158 @@
+import { ethers, deployments, network } from "hardhat"
+import { BigNumber } from "ethers"
+import fetch from "node-fetch"
+import { MiniChefV2, SDL } from "../build/typechain"
+
+interface TimeToRateMap {
+  [timestamp: number]: BigNumber
+}
+
+async function main() {
+  const latestBlock = await ethers.provider.getBlock("latest")
+  const latestBlockTimestamp = latestBlock.timestamp
+
+  const etherscanAPIUrl = network.verify?.etherscan?.apiUrl
+  const etherscanAPIKey = network.verify?.etherscan?.apiKey
+  if (!etherscanAPIUrl) {
+    throw new Error(
+      `No etherscan API URL found in hardhat.config.js file for network ${network.name}`,
+    )
+  }
+
+  const minichef = (await ethers.getContract("MiniChefV2")) as MiniChefV2
+  const creationTxHash = (await deployments.get("MiniChefV2"))
+    .transactionHash as string
+  const creationBlockNumber = (
+    await ethers.provider.getTransaction(creationTxHash)
+  ).blockNumber as number
+  const creationBlockTimestamp = (
+    await ethers.provider.getBlock(creationBlockNumber)
+  ).timestamp
+
+  // Check current Saddle per second rate
+  const currentSaddleRate = await minichef.saddlePerSecond()
+  if (currentSaddleRate.gt(0)) {
+    console.warn(
+      "\x1b[33m%s\x1b[0m",
```

Comment:
what is this?

Reply:
It lets you print in yellow using some terminal magic